spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/SparkV2Filters.java

```diff
@@ -289,11 +311,23 @@ private static Object convertLiteral(Literal<?> literal) {
     return literal.value();
   }
 
-  private static Expression handleEqual(String attribute, Object value) {
-    if (NaNUtil.isNaN(value)) {
-      return isNaN(attribute);
+  private static UnboundPredicate<Object> handleEqual(UnboundTerm<Object> term, Object value) {
+    if (value == null) {
+      return isNull(term);
+    } else if (NaNUtil.isNaN(value)) {
+      return isNaN(term);
+    } else {
+      return equal(term, value);
+    }
+  }
+
+  private static UnboundPredicate<Object> handleNotEqual(UnboundTerm<Object> term, Object value) {
+    if (value == null) {
+      return notNull(term);
```

Comment:
`<>` is not null safe right? So this should return `null` or throw an exception if the value is null.

Reply:
Sorry for missed that.